File: webapp/src/services/ArrivalTime.js
class ArrivalTime {
    constructor(lineName, destinationName, expectedSeconds) {
        this._lineName = lineName;
        this._destinationName = destinationName;
        this._expectedSeconds = expectedSeconds;
    }

    get lineName() {
        return this._lineName;
    }

    get expectedSeconds() {
        return this._expectedSeconds;
    }

    get destinationName() {
        return this._destinationName;
    }
}

export default ArrivalTime;

File: backend/train/LDB/LDB/Model/ArrivalTime.cs
namespace LDB.Model
{
    public sealed class ArrivalTime
    {
        public ArrivalTime(string atStationCode, string destinationCode, string lineName, int expectedSecs, string expectedTime)
        {
            AtStationCode = atStationCode;
            DestinationCode = destinationCode;
            LineName = lineName;
            ExpectedSecs = expectedSecs;
            ExpectedTime = expectedTime;
        }

        /// <summary>
        /// Station code we are interested in.
        /// </summary>
        public string AtStationCode { get; }

        /// <summary>
        /// Final destination code.
        /// </summary>
        public string DestinationCode { get; }

        /// <summary>
        /// The train operator for that train.
        /// </summary>
        public string LineName { get; }

        /// <summary>
        /// Expected arrival in Seconds.
        /// </summary>
        public int ExpectedSecs { get; }

        /// <summary>
        /// The expected timere returned by datasource.
        /// </summary>
        public string ExpectedTime { get; }
    }
}
